src/controllers/api.js

```diff
@@ -34,7 +34,24 @@ async function fetchCompressedConfiguration(config) {
   }
 }
 
-async function fetchResults() {
+async function fetchPartMC() {
+  try {
+    const response = await axios.get(`${apiUrl}/api/download-partmc`, {
+      params: {},
+      responseType: 'arraybuffer'
+    });
+    const blob = new Blob([response.data], {
+      type: response.headers.get("content-type"),
+    });
+    return window.URL.createObjectURL(blob);
+  } catch (error) {
+    console.error(`Error fetching compressed configuration: ${error.message}`);
+    throw error;
+  }
+}
```

Comment:
This functionality should be handled by `download-results`. See my comments below

```suggestion
```